pkg/controller/labelidentity/label_group_index_test.go

```diff
@@ -599,24 +598,29 @@ func TestDeleteLabelIdentity(t *testing.T) {
 			i.AddLabelIdentity(labelA, 1)
 			i.AddLabelIdentity(labelB, 2)
 			i.AddLabelIdentity(labelC, 3)
+			assert.EventuallyWithT(t, func(t *assert.CollectT) {
+				lock.Lock()
+				defer lock.Unlock()
+				assert.ElementsMatch(t, actualPoliciesCalled, []string{"policyA", "policyB", "policyA", "policyB", "policyD", "policyA", "policyB"}, "Unexpected policy sync calls")
+			}, time.Second, 10*time.Millisecond)
 			// Reset the actualPoliciesCalled slice. Otherwise, the list will be filled with extra items
 			// in the channel from preloading label identities.
-			time.Sleep(10 * time.Millisecond)
 			lock.Lock()
```

Comment:
nit: I'm assuming that with `assert.Eventually` above this lock is no longer necessary cause we always wait until the preloads to complete and lock released or the test fails. But I have no problem keeping it just to be safe

Reply:
You never know, if there is a bug in the controller I imagine it could fire an extra call to the even handler. It's better to be safe and I prefer a "normal" test failure to an error reported by the race detector. This is also why I added calls to `i.lock.Lock()` when accessing `i.labelIdentities`.